sonar-core/src/main/java/org/sonar/core/user/GroupMembershipDao.java

```diff
@@ -58,6 +63,22 @@ public int countGroups(SqlSession session, GroupMembershipQuery query, Long user
     return mapper(session).countGroups(params);
   }
 
+  public Multimap<String, String> selectGroupsByLogins(final DbSession session, Collection<String> logins) {
+    final Multimap<String, String> result = ArrayListMultimap.create();
+    DaoUtils.executeLargeInputs(logins, new NonNullInputFunction<List<String>, List<LoginGroup>>() {
```

Comment:
we should not use NonNullInputFunction anymore, if my understanding is correct but the @NonNull annotation to override the Function contract